### PR TITLE
[Method References 2] LPS-167249 Convert method reference to ServiceTracker in module portal-search-web

### PR DESCRIPTION
**Pull request by W2jx**

### CODE REVIEW

**Comment by zhangyan0701** [comment]
ci:test:search

**Comment by Kvale1733** [comment]
@W2jx No regressions found.

**Comment by W2jx** [comment]
Force-Push is just to resolved the conflict about file SearchFacetTracker.java.

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
ci:test:sf

**Comment by W2jx** [comment]
ci:test:search

**Comment by catarineaguiar** [comment]
@W2jx no regressions found.

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/facet/SearchFacetRegistry.java [hunk 1]
```diff
@@ -35,20 +38,50 @@ public List<SearchFacet> getSearchFacets() {
...
 	}
 
-	@Reference(
-		cardinality = ReferenceCardinality.MULTIPLE,
-		policy = ReferencePolicy.DYNAMIC,
-		policyOption = ReferencePolicyOption.GREEDY
-	)
-	protected void addSearchFacet(SearchFacet searchFacet) {
-		_searchFacets.add(searchFacet);
+	@Activate
```
Comment (tinatian) [inline modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/facet/SearchFacetRegistry.java:41]: @W2jx Please use ServiceTrackerList

**Comment by tinatian** [comment]
@W2jx Please see my comment and resend.

Tina

**Pull request closed without merging**